python/GafferUI/_StyleSheet.py

```diff
@@ -113,14 +113,23 @@
 	# control will not be portable across different backgrounds.
 
 	"tintLighterSubtle" :   ( 255, 255, 255, 10 ),
-	"tintLighter" :         ( 80, 80, 80, 255 ),
+	"tintLighter" :         ( 255, 255, 255, 20 ),
 	"tintLighterStrong" :   ( 255, 255, 255, 40 ),
 	"tintLighterStronger" : ( 255, 255, 255, 100 ),
-	"tintDarkerSubtle" :    ( 80, 80, 80, 255 ),
+	"tintDarkerSubtle" :    ( 0, 0, 0, 10 ),
 	"tintDarker" :          ( 0, 0, 0, 20 ),
 	"tintDarkerStrong" :    ( 0, 0, 0, 40 ),
 	"tintDarkerStronger" :    ( 0, 0, 0, 70 ),
 
+	# Borders within controls need to sit within the colors of the control, not
+	# the colors of the background.  There are situations where using
+	# transparency would show through to the background behind the control, not
+	# the background of the control, in which case it's better to use an opaque
+	# tint, relative to the standard grey we usually use for control backgrounds.
+	"internalTintLighter" :  ( 88, 88, 88 ),
+	"internalTintDarker" : ( 78, 78, 78 ),
```

Comment:
> My first premise: using transparency for adjoined edges has never really been fulfilling its intended function. It would make sense if the edges were overlayed over the adjoined widgets background - but that's not how it's been working, it's showing through to whatever is behind the widget.

Makes sense.

> Based on this, my goal is just to choose opaque colors to use for adjoined edges which recreate the current appearance of adjoined edges. I'm looking at the play/stop buttons on InteractiveArnoldRender.

Also makes sense. I spent a bit of time googling to see if we could get Qt to draw the borders over the background, but I didn't come up with anything.

> After setting appropriate opaque colors, the appearance is pretty close, but the 2 pixels underneath the edge take the edge colors rather than blending into the bottom edge.

That's close enough for me.

I am a bit unhappy at the number of named colours in this stylesheet now though - it's hard to know what to use and when, and the use-transparency-for-tinting concept is creaking in yet another place. I have a few proposals there, none of them great :

1. Stick with introducing two new colours as you've done. But call them `backgroundDarkHighlightSubtle` and `backgroundLightHighlightSubtle` and put them with the original `backgroundDarkHighlight` and  $backgroundLightHighlight` colours which form the border edges for non-adjoining buttons. I hope by placing them with their associated values and using the same `Subtle` terminology as elsewhere might make this a tiny bit more understandable?
2. Don't modify the colours for adjoining buttons at all. This makes the join harder, and is less visually pleasing, but avoids yet more complexity in the stylesheet (picture below).
3. Embark on a simplification of the stylesheet based on an acceptance that tints don't work as well as we'd like, with the goal of reducing the complexity of the thing. Obviously this isn't an option for now.

![image](https://user-images.githubusercontent.com/1133871/186893876-54341153-beca-4ba0-973c-70afb46d4b5f.png)

Reply:
Actually, thinking further, I don't like either the "Subtle" or "Internal" modifiers for these colours. The former doesn't give you any clue what the purpose of the colour is, and the latter does but deviates from the terminology we're already using. So I think we should use "Adjoined" instead, making it completely clear what it is for. In which case they would be `backgroundDarkHighlightAdjoined` and `backgroundLightHighlightAdjoined`. 

I don't know why the button uses `backgroundDarkHighlight` instead of `backgroundLightLowlight` when the documentation suggests that you use `<backgroundType><Highlight|Lowlight>` and `backgroundType` here would be `Light` according to the other colour assignments for the button. I guess it's just that it looks nicer that way...

Reply:
> 3. Embark on a simplification of the stylesheet based on an acceptance that tints don't work as well as we'd like

Obviously we're not doing this now, but I think one principal for this might be that tinting should never be used for content widgets (buttons, text fields, tree views, other things you interact with). So those all have the same visual appearance regardless of nesting. And then tinting only gets used on backgrounds to indicate nesting - tabbed containers, frames and the like. Maybe that was the original idea anyway, and the buttons were just an anomaly?

Reply:
I've gone with backgroundDarkHighlightAdjoined in b2ca2ad

I also went back to 0px corners, since it seems like the 1px ones might look weird at high DPI, and you said you were OK with how the 0px ones looked.